### DISCUSSION TITLE
Should I be able to access all OSS-Fuzz reports through the emailed links?

### DISCUSSION THREAD

**Discussion by EliahKagan** [gitpython-developers/GitPython#1947] [Q&A] [answered]
@DaveLak I wanted to let you know about this, but it is definitely not at all urgent.

Although I receive emails about OSS-fuzz issues for GitPython, I'm not actually able to access the linked issues in the tracker to see full details, until they are made public. I have made sure I am signing in with the correct Google account. Based on https://github.com/gitpython-developers/GitPython/discussions/1889 I think this is unintended.

I am guessing the problem may be that the configured email address for me starts `eliah.kagan`. While that's the form of that email address that I prefer, technically Gmail considers `eliahkagan`, without the `.`, to be its preferred form. Because this integrates with the tracker--a Google email address had to be used--my guess is that it may be the `.` that is keeping me out.

Per https://github.com/gitpython-developers/GitPython/discussions/1889#discussioncomment-9280557, I've opened this as a new discussion (it's not really a bug in GitPython, after all). This too can be closed once resolved.

**Comment by Byron** [comment]
As a digression of the original topic: I also do get those emails but ignore them. However, I wonder if those are something that should be fixed, or can be fixed at all.

**Reply by EliahKagan** [reply]
Are you also unable to access the linked issues until they are made public?

**Reply by Byron** [reply]
Yes, that works for me (but I have to use a Chomium based browser to be able to log-in.

**Reply by EliahKagan** [reply]
The browser thing seems odd, but I have been using Chrome for this, so that's not the cause of the problem I'm having.

**Reply by DaveLak** [reply]
Sorry @Byron, that noise is my fault. You don't need to bother looking at the issues logged right now; they aren't true bugs, but rather explicitly raised GitPython exceptions that I failed to account for in the test harness.

I apologize for not resolving them sooner. I have a local branch with a solution about 90% complete, but I've had life get in the way of completing the PR to fix it recently.

I haven't forgotten or abandoned it though. I intend to get a fix up for review in the coming week

**Reply by DaveLak** [reply]
The PR I was referring to above is up! https://github.com/gitpython-developers/GitPython/pull/1950

That should (🤞) resolve all of those noisy OSSFuzz issues being logged

**Comment by DaveLak** [comment]
@EliahKagan Yes, you should be able to access the private issues in the tracker (via the links in the emails, or via your account dashboard) once authenticated via the "Sign in with a Google account" option on the issue tracker (the tracker is named Monorail) using the email I added to the config in the OSS-Fuzz repo. There appears to be a GitHub SSO option as well, but I've never got it to work for me.

I'm not sure if the `.` in the name makes any difference but it's worth a shot to try logging in without it. My email has two periods and works, but I registered the email with them so it's not a "dot alias" like I know Google allows. I'd expect that "plus aliases" wouldn't work (e.g. something+email-address@google.com), so maybe dot aliases are similar?

We can update the config in the upstream repo if you'd like.

**Reply by EliahKagan** [reply]
> I'm not sure if the `.` in the name makes any difference but it's worth a shot to try logging in without it.

I think I am always logged in without it, in the sense that only the dotless form is ever available to select in "Sign in with a Google account" and that emails generated from GMail, such as security alerts, always show it without the dot.

[bugs.chromium.org](https://bugs.chromium.org/) shows the user https://bugs.chromium.org/u/1678564301/ as having never signed in.

[oss-fuzz.com](https://oss-fuzz.com/), as well as specific pages such as https://oss-fuzz.com/testcase-detail/4919835776057344, say:

```text
You (email=eliahkagan@gmail.com) are not authorized to access this page!
```

It looks like whether dots are significant in Gmail addresses [depends on whether the address is personal or organizational](https://support.google.com/mail/answer/7436150) but that Gmail does not consider them significant in `@gmail.com` addresses. Whether that's at all relevant to this, I don't know.

I guess let's try changing my email address for it so it doesn't have the dot and is the address shown in the "not authorized" message above.

**Reply by DaveLak** [reply, marked as answer]
> [oss-fuzz.com](https://oss-fuzz.com/), as well as specific pages such as [oss-fuzz.com/testcase-detail/4919835776057344](https://oss-fuzz.com/testcase-detail/4919835776057344), say:
>```
>You (email=eliahkagan@gmail.com) are not authorized to access this page!
>```

Thanks for sharing the error message. Seeing as the email it displays has no `.`, I think that very well may be the issue.

I put up a quick patch PR to update the email in the OSS-Fuzz repo: https://github.com/google/oss-fuzz/pull/12281

Hopefully that's all it'll take to fix your access.

**Reply by EliahKagan** [reply]
Thanks!

**Reply by EliahKagan** [reply]
The change in https://github.com/google/oss-fuzz/pull/12281 has been merged. But I still get the same error message at https://oss-fuzz.com/ and https://oss-fuzz.com/testcase-detail/4919835776057344.

The user account I am signed in with on bugs.chromium.org is https://bugs.chromium.org/u/3928863871/ with the email address of `eliahkagan@gmail.com` shown. The user account that starts with `eliah` and is authorized to access the reports is https://bugs.chromium.org/u/1678564301/. I suspect this has something to do with the problem. I cannot view the email address associated with that account; I had assumed it was `eliah.kagan@gmail.com` or `eliahkagan@gmail.com` and suspected it was the latter.

Maybe there are two separate bugs.chromium.org accounts associated somehow with the same email address? But even if so, I wouldn't know how to sign in as the other account.

**Reply by DaveLak** [reply]
The changes merged into https://github.com/google/oss-fuzz aren't deployed immediately. I'm not sure how long it'll take for the authentication to use the updated config but I'd expect 12 to 48 hours.

You can see the build logs here: https://oss-fuzz-build-logs.storage.googleapis.com/index.html#gitpython

**Reply by EliahKagan** [reply]
That's great--thanks for the info! I'll let you know how it goes after the next build.

**Reply by EliahKagan** [reply]
Although it doesn't look like there's been another build yet, I received additional emails about the open issues, and the links--including the oss-fuzz.com and bugs.chromium.org links given above, are working now. I seem to be able to access everything.

Thanks again for you help!
